prombench won't build outside $GOPATH when using the docker image

prombench won't build outside `$GOPATH` when using the docker image **if** there's no `/go/src/github.com/prometheus/prombench`

docker golang was used to build prombench was added here:
https://github.com/prometheus/prombench/pull/185/files#diff-b67911656ef5d18c4ae36cb6741b7965R4

When running in a machine with `GO111MODULE='on'` , `go build ./cmd/prombench` works outside `$GOPATH`  but running  `make build` which uses the docker golang image does not work unless prombench is cloned at  `/go/src/github.com/prometheus/prombench`.

If the `PROMBENCH_DIR = /go/src/github.com/prometheus/prombench` is changed to `PROMBENCH_DIR = $(shell pwd)` in `/Makefile` which will point to the prombench directory outside of `$GOPATH` it errors out:

```
λ make build                                                                                                                              🚸[master]
go: cannot find main module; see 'go help modules'
make: *** [Makefile:47: build] Error 1
```

but this problem is specifically with the `go build` command in the docker image, If I replace the `go build ./cmd/prombench` to `go mod verify`  it works as expected, indicating it is pointing to the correct directory with the `go.mod` file.

I am not  sure why `make build` is failing when changing `PROMBENCH_DIR = $(shell pwd)`. 

cc @krasi-georgiev

--- a/Makefile
+++ b/Makefile
@@ -1,15 +1,15 @@
 PROMBENCH_CMD        = ./prombench
 DOCKER_TAG = docker.io/prombench/prombench:2.0.0
 GOLANG_IMG = golang:1.12
-PROMBENCH_DIR = /go/src/github.com/prometheus/prombench
 USERID = $(shell id -u ${USER})
 USERGROUP = $(shell id -g ${USER})
 DOCKER_CMD = docker run --rm \
 			  -e GOPATH='/go' \
 			  -e GO111MODULE='on' \
 			  -e GOCACHE='/tmp/.cache' \
+			  -v ${PWD}:/prombench \
 			  -v ${GOPATH}:/go \
-			  -w $(PROMBENCH_DIR) \
+			  -w /prombench \
 			  -u $(USERID):$(USERGROUP) \
 			  $(GOLANG_IMG)
 

